pkg/prometheus/operator.go

```diff
@@ -2590,6 +2595,24 @@ func validateRelabelConfig(rc monitoringv1.RelabelConfig) error {
 	if rc.Action == string(relabel.Replace) && !relabelTarget.MatchString(rc.TargetLabel) {
 		return errors.Errorf("%q is invalid 'target_label' for %s action", rc.TargetLabel, rc.Action)
 	}
+
+	// uppercase and lowercase actions are available from Prometheus 2.36.0 only
+	if ((rc.Action == string(relabel.Lowercase) || rc.Action == string(relabel.Uppercase)) && minimumVersion) && !relabelTarget.MatchString(rc.TargetLabel) {
```

Comment:
if we check for version compatibility first, we could omit `minimumVersion` here.